### ISSUE TITLE
server loses part of the initial flight, if ClientHello is split across multiple packets

### ISSUE THREAD

**Issue by marten-seemann** [quic-go/quic-go#3760]
I observed this when running the new post-quantum test, while disabling packet loss (https://github.com/quic-go/quic-go/pull/3759).

<img width="1511" alt="image" src="https://user-images.githubusercontent.com/1478487/230010956-245b4c18-5fde-44fa-9242-d52c9ed3a7b4.png">

**Comment by marten-seemann** [comment]
```
2023/04/05 18:19:18.148599 server <- Received Initial packet.
2023/04/05 18:19:18.148620 server Changing connection ID to f63d36e5.
2023/04/05 18:19:18.148723 muxer Adding connection IDs 122ab6426e54b31d1858f257323cc6 and f63d36e5 for a new connection.
2023/04/05 18:19:18.148730 server <- Received Initial packet.
2023/04/05 18:19:18.148732 server Changing connection ID to 5960239e.
2023/04/05 18:19:18.148734 muxer Not adding connection ID 122ab6426e54b31d1858f257323cc6 for a new connection, as it already exists.
```

**Comment by marten-seemann** [comment]
This is not good. The problem is that we're queuing the packets with the server as soon as we receive them, and _then_ the server processes these packets, without doing another lookup in the CID map.

**Comment by marten-seemann** [comment]
Removing from the v0.34 milestone. While annoying, this doesn't happen in practice unless PQ is used (since otherwise there's at least a few hundred milliseconds between sending of the initial Initial and its retransmission). And even in the PQ case, this is recoverable via PTO. Not ideal, but doesn't require an immediate fix either.

This might be easier to fix after the giant API refactor (#3727).
